### PR TITLE
Allow incloming correlation_id propagation

### PR DESCRIPTION
**Pull request by ilipkind**
Making sure that incoming correlation id header is used if provided

### CODE REVIEW

File: src/test/java/com/ge/predix/log/filter/LogFilterTest.java [hunk 1]
```diff
@@ -129,4 +132,26 @@ public void testConstructorArgumentsNoDefaultZone() {
...
+    @Test
+    public void testLogFilterWithCorrelationIdHeader() throws Exception {
+        LogFilter logFilter = new LogFilter();
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        String requestCorrelationId = UUID.randomUUID().toString();
+        request.addHeader("Correlation-Id", requestCorrelationId);
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        logFilter.doFilter(request, response, new MockFilterChain());
+        String resposnseCorrelationId = response.getHeader("Correlation-Id");
+        Assert.assertEquals(requestCorrelationId, resposnseCorrelationId);
```
Comment (DarioAmiri) [inline src/test/java/com/ge/predix/log/filter/LogFilterTest.java]: Typo. Misspelled "response"

File: src/main/java/com/ge/predix/log/filter/LogFilter.java [hunk 2]
```diff
@@ -87,8 +87,11 @@ public LogFilter() {
     @Override
     protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response,
             final FilterChain filterChain) throws ServletException, IOException {
-
-        String correlationId = UUID.randomUUID().toString();
+    	String correlationId = request.getHeader(LOG_CORRELATION_ID);
```
Comment (DarioAmiri) [inline src/main/java/com/ge/predix/log/filter/LogFilter.java]: Formatting is off.

File: src/main/java/com/ge/predix/log/filter/LogFilter.java [hunk 3]
```diff
@@ -87,8 +87,11 @@ public LogFilter() {
...
+        if (StringUtils.isEmpty(correlationId)) {
+                correlationId = UUID.randomUUID().toString();
```
Comment (DarioAmiri) [inline src/main/java/com/ge/predix/log/filter/LogFilter.java]: Formatting is off. Recommend importing the ACS project format and code cleanup configurations.

File: src/test/java/com/ge/predix/log/filter/LogFilterTest.java [hunk 4]
```diff
@@ -129,4 +132,26 @@ public void testConstructorArgumentsNoDefaultZone() {
...
+    }
+
+    @Test
+    public void testLogFilterWithoutCorrelationIdHeader() throws Exception {
+        LogFilter logFilter = new LogFilter();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        logFilter.doFilter(new MockHttpServletRequest(), response, new MockFilterChain());
+        String resposnseCorrelationId = response.getHeader("Correlation-Id");
+        Assert.assertNotNull(resposnseCorrelationId);
```
Comment (DarioAmiri) [inline src/test/java/com/ge/predix/log/filter/LogFilterTest.java]: Typo.

File: src/main/java/com/ge/predix/log/filter/LogFilter.java (hunk 2 above)
Comment (ilipkind) [inline src/main/java/com/ge/predix/log/filter/LogFilter.java]: fixed

**Pull request merged**